### PR TITLE
Provide a tool to copy data between Cassandra clusters

### PR DESCRIPTION
**Pull request by shanson7**
This is similar to `mt-update-ttl` but allows the destination cluster to differ from the source.

Additionally, it uses unlogged batches for performance.

### CODE REVIEW

File: cmd/mt-store-cp/main.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+	startTs      = flag.Int("start-timestamp", 0, "timestamp at which to start, defaults to 0")
+	endTs        = flag.Int("end-timestamp", math.MaxInt32, "timestamp at which to stop, defaults to int max")
+	startToken   = flag.Int64("start-token", minToken, "token to start at (inclusive), defaults to math.MinInt64")
+	endToken     = flag.Int64("end-token", maxToken, "token to stop at (inclusive), defaults to math.MaxInt64")
+	numThreads   = flag.Int("threads", 1, "number of workers to use to process data")
+	maxBatchSize = flag.Int("max-batch-size", 10, "max number of queries per batch")
+
+	idxTable   = flag.String("idx-table", "metric_idx", "idx table in cassandra")
+	partitions = flag.String("partitions", "*", "process ids for these partitions (comma separated list of partition numbers or '*' for all)")
```
Comment (Dieterbe) [inline cmd/mt-store-cp/main.go]: might want to add validation that this should not be an empty string, or simply that it is a series of digits (can be parsed with strconv.Atoi)

File: cmd/mt-store-cp/main.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+			doneKeysSnap := atomic.LoadUint64(&doneKeys)
+			doneRowsSnap := atomic.LoadUint64(&doneRows)
+			fmt.Fprintf(os.Stderr, "ERROR: id=%d failed querying %s: %q. processed %d keys, %d rows\n", id, tableIn, err, doneKeysSnap, doneRowsSnap)
+		}
+		atomic.AddUint64(&doneKeys, 1)
+	}
+}
+
+func update(sourceSession, destSession *gocql.Session, tableIn, tableOut string) {
+	// Get the list of ids that we carry about
```
Comment (Dieterbe) [inline cmd/mt-store-cp/main.go]: care about?

**Comment by Dieterbe** [comment]
do you remember offhand what kind of improvement you got thanks to the unlogged batches?

this tool looks fit for merging.was it reliable in your experience?

**Comment by shanson7** [comment]
Offhand, I don't know. While running it in k8s it was unfortunately unreliable. After some time, writes would stop being reflected (although gocql indicated they were succeeding). Not sure if that was our Cassandra setup or not though. Either way, it takes days to copy even a small time range :/

**Comment by shanson7** [comment]
Without unlogged batches, it was far slower, but exhibited the same behavior of writes just stopping. Very strange.

**Comment by Dieterbe** [comment]
then I think i shall merge it but mark it as experimental.

File: cmd/mt-store-cp/main.go (hunk 1 above)
Comment (Dieterbe) [reply cmd/mt-store-cp/main.go]: never mind i see now that you do this.

**Comment by Dieterbe** [comment]
merging via #909 
thanks @shanson7 !

**Pull request closed without merging**